app/models/thredded/topic.rb

```diff
@@ -15,6 +15,23 @@ class Topic < ActiveRecord::Base
 
     scope :order_sticky_first, -> { order(sticky: :desc) }
 
+    scope :followed_by, lambda { |user|
+      joins('INNER JOIN thredded_user_topic_follows ON thredded_user_topic_follows.topic_id = thredded_topics.id')
+        .where('thredded_user_topic_follows.user_id = ?', user.id)
+    }
+    scope :unread_by, lambda{ |user|
+      joins(<<-SQL)
```

Comment:
It's best to do this with AREL instead of custom SQL fragments, because arel queries are easier to refactor because they are composable, there is less scope for errors, and the generated SQL will be more compact for the specific database. See here for an example of how `LEFT JOIN` can be done with Arel: https://github.com/thredded/thredded/blob/8772b099863c441f5716285a44fb8648517eb083/app/models/concerns/thredded/topic_common.rb#L44